### PR TITLE
Add Face Recognition 1:1 Example

### PR DESCRIPTION
**Pull request by munkyshi**
### Linked issue(s)
Fixes KOL-6706

### What change does this PR introduce and why?
Adds code for Face Recognition 1:1 Example

### Please check if the PR fulfills these requirements

- [ ] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Relevant tests for the changes have been added
- [ ] Relevant docs have been added / updated

### CODE REVIEW

**Comment by munkyshi** [comment]
Example integration: https://trunk.kolena.com/andrew/dataset/standards?datasetId=150

File: examples/dataset/face_recognition_11/face_recognition_11/image.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from dataclasses import dataclass
+
+from kolena.asset import ImageAsset
+
+
+@dataclass(frozen=True)
+class FRImageAsset(ImageAsset):
```
Comment (munkyshi) [inline examples/dataset/face_recognition_11/face_recognition_11/image.py]: TODO: Is this needed?

**Comment by y27choi** [comment]
Nice integration! Here are some comments from reviewing the example:
* Should we add similarity threshold information somewhere? 
image

* Have you considered using eval config for different FMR threshold?
* Curious to see why you decided to go with ImageAssets instead of the Composite type.

**Comment by munkyshi** [comment]
> Should we add similarity threshold information somewhere?

Good call, will do.

> Have you considered using eval config for different FMR threshold?

That was in the original example, but it makes it difficult to actually see how things perform at different FMRs at once if every FMR is a different model + eval config.
Screenshot 2024-06-14 at 12 46 44
In general, I think many of our original eval config use cases could have been metrics groups.

> Curious to see why you decided to go with ImageAssets instead of the Composite type.

Didn't we stop supporting composites? I went with a pair of image assets because the only way we have to view two images (with annotations) at once was with a list of assets.

File: examples/dataset/face_recognition_11/README.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
+# Example Integration: Face Recognition (1:1)
+
+This example integration uses the [Labeled Faces in the Wild (LFW)](http://vis-www.cs.umass.edu/lfw/)
+dataset and Face Recognition (FR) workflow to
+demonstrate how to test and evaluate end-to-end FR (1:1) model pipelines on Kolena.
```
Comment (y27choi) [inline examples/dataset/face_recognition_11/README.md:5]: We might want to update this now that we are focusing only on the FR evaluation.

File: examples/dataset/face_recognition_11/README.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+Once the dataset and results have been uploaded to Kolena, visit [Kolena](https://app.kolena.io/redirect/) to
+[explore the data and results](https://docs.kolena.com/dataset/quickstart/#step-3-explore-data-and-results).
+
+Here are our [Quality Standards](https://docs.kolena.com/dataset/core-concepts/quality-standard/) recommendations for
+face recognition [1:1]:
+
+### Metrics
+
+1. F1 Score [detection]
```
Comment (y27choi) [inline examples/dataset/face_recognition_11/README.md]: Should update these as well

File: examples/dataset/face_recognition_11/face_recognition_11/upload_dataset.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+def run(args: Namespace) -> None:
+    df_metadata = pd.read_csv(DATASET_METADATA)
+    df_pairs = pd.read_csv(DATASET_PAIRS)
+    df_detection = pd.read_csv(DATASET_DETECTION)
+
+    genuine_pairs = {}
+    imposter_pairs = {}
+    for record in df_pairs.itertuples():
+        if record.is_match:
+            genuine_pairs[record.locator_1] = record.locator_2
```
Comment (y27choi) [inline examples/dataset/face_recognition_11/face_recognition_11/upload_dataset.py]: Should we rename these locators to `locator_left` and `locator_right` to be consistent with the left / right usage?

**Review by y27choi — APPROVED** [review]
Left some nitpicky comments but overall it looks very clean and addresses all the issues that listed in the ticket.

Just checking, have we updated the non-raw csv files on the s3 bucket?

**Pull request merged**